### ISSUE TITLE
HTTP Error 404 by Download from TVNow

### ISSUE THREAD

**Issue by cannamaier** [rg3/youtube-dl#18499]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.12.09*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.12.09**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [-v', 'https://www.tvnow.de/shows/grill-den-henssler-2471']
[debug] Encodings: locale cp1252, fs mbcs, out cp866, pref cp1252
[debug] youtube-dl version 2018.12.09
[debug] Python version3.4.4 (C Python) - Windows-8.1-6.3.9600
[debug] exe versions: none
[debug] Proxy map: {}
[TVNowShow] grill-den-henssler-2471: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HT
TPError()); please report this issue on https://yt-dl.org/bug . Make sure you ar
e using the latest version; type  youtube-dl -U  to update. Be sure to call yout
ube-dl with the --verbose flag and include its complete output.
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmpg8efod20\bu
ild\youtube_dl\extractor\common.py", line 605, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmpg8efod20\bu
ild\youtube_dl\YoutubeDL.py", line 2212, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 580, in http_response
  File "C:\Python\Python34\lib\urllib\request.py", line 508, in error
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Python\Python34\lib\urllib\request.py", line 588, in http_error_default
<end of log>
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):

- Playlist: https://www.tvnow.de/shows/grill-den-henssler-2471
- Single-Video: https://www.tvnow.de/shows/buschi-vs-koeppen-15474/staffel-1/episode-2-samba-in-rio-1327175

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/rg3/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.
If work on your *issue* requires account credentials please provide them or explain how one can obtain them.

**Comment by DerBunteBall** [comment]
Can agree. I accidentally opened also an issue in the same time window. See #18500.

The portal cahnged totally today or yesterday. The extractor seems to need rework again. As we know TVNow often changes.

Best Regards

**Comment by fabki** [comment]
Should not be a major re-programming part.

API changed from
https://api.tvnow.de/v3/
to
https://apigw.tvnow.de/

**Comment by revunix** [comment]
@fabki good to know, have you already tested it?

**Comment by fabki** [comment]
not yet, maybe after work on the late hours today.

**Comment by revunix** [comment]
That would be nice. 👊🏻

**Comment by fabki** [comment]
Hmm seems not to be so easy as i thought.

Only free content can be grabed with new api, but script also needs modification (i'm not good with python :( ). The DRM-free "Premium-Content" can not be accessed without credentials. (i.e. older episodes of free content)

The "good" one:
Simple need to grab the ID - i.e. 1333774 -> https://apigw.tvnow.de/module/player/**1333774** - and getting all informations that we need to format and download.

**Comment by revunix** [comment]
I tried it out, the dashhd works ... and i can it download with the old version of youtube-dl. Thats good so far. Its more work but it works for now.

**Comment by revunix** [comment]
hmm still not updated? :(

**Comment by revunix** [comment]
1080p is not available.

```
python -m youtube_dl https://www.tvnow.de/serien/gzsz-1/2019-01/episode-6671-leon-ist-entsetzt-1354104 -F
[TVNow] gzsz/leon-ist-entsetzt: Downloading JSON metadata
[TVNow] 1354104: Downloading MPD manifest
[TVNow] 1354104: Downloading ISM manifest
[TVNow] 1354104: Downloading m3u8 information
[info] Available formats for 1354104:
format code         extension  resolution note
mss-audio-128       isma       audio only  128k , AACL  (48000Hz)
dash-audio=128000   m4a        audio only [de] DASH audio  128k , m4a_dash container, mp4a.40.2 (48000Hz)
hls-136             mp4        audio only  136k , mp4a.40.2@128k
mss-video-500       ismv       640x360     500k , AVC1, video only
dash-video=500000   mp4        640x360    [en] DASH video  500k , mp4_dash container, avc1.4D001F, 25fps, video only
mss-video-1000      ismv       640x360    1000k , AVC1, video only
dash-video=1000000  mp4        640x360    [en] DASH video 1000k , mp4_dash container, avc1.4D001F, 25fps, video only
mss-video-1500      ismv       960x540    1500k , AVC1, video only
dash-video=1500000  mp4        960x540    [en] DASH video 1500k , mp4_dash container, avc1.4D001F, 25fps, video only
mss-video-3000      ismv       960x540    3000k , AVC1, video only
dash-video=3000000  mp4        960x540    [en] DASH video 3000k , mp4_dash container, avc1.4D001F, 25fps, video only
hls-666             mp4        640x360     666k , avc1.77.31@ 500k, 25.0fps, mp4a.40.2@128k
hls-1196            mp4        640x360    1196k , avc1.77.31@1000k, 25.0fps, mp4a.40.2@128k
hls-1726            mp4        960x540    1726k , avc1.77.31@1500k, 25.0fps, mp4a.40.2@128k
hls-3316            mp4        960x540    3316k , avc1.77.31@3000k, 25.0fps, mp4a.40.2@128k (best)
```

for example: 
https://vodnowusodash.secure.footprint.net/proxy195/clear/ngvod/tvnow/1354104-1-15729.ism/.mpd?filter=(type%3d%3d%22audio%22)%7c%7c(type%3d%3d%22video%22%26%26systemBitrate%3c8616000)

**Comment by krisrok** [comment]
issue persists but is closed.
